easybuild/easyconfigs/r/RAxML-NG/RAxML-NG-0.9.0-gompi-2019b.eb

```diff
@@ -0,0 +1,87 @@
+easyblock = 'CMakeMakeCp'
+
+name = 'RAxML-NG'
+version = '0.9.0'
+
+local_terrap_tree = '6dcb061ceb6ed41a7ebe1a8af1f3354e0126125c'
+local_pllmod_tree = '840a19525ed133ad243afb23124911b849d4685d'
+local_libpll_tree = '6874327c9ca9544e27b426c0e7205c8b6f4ed297'
+
+homepage = "https://github.com/amkozlov/raxml-ng"
+description = """RAxML-NG is a phylogenetic tree inference tool which
+uses maximum-likelihood (ML) optimality criterion. Its search heuristic
+ is based on iteratively performing a series of Subtree Pruning and Regrafting
+ (SPR) moves, which allows to quickly navigate to the best-known ML tree."""
+
+toolchain = {'name': 'gompi', 'version': '2019b'}
+
+sources = [
+    {
+        'source_urls': ['https://github.com/amkozlov/raxml-ng/archive/'],
+        'filename': '%(version)s.tar.gz',
+    },
+    {
+        'source_urls': ['https://github.com/amkozlov/terraphast-one/archive/'],
+        'download_filename': local_terrap_tree + '.zip',
+        'filename': 'terraphist-one.zip',
+    },
+    {
+        'source_urls': ['https://github.com/ddarriba/pll-modules/archive/'],
+        'download_filename': local_pllmod_tree + '.zip',
+        'filename': 'pll-modules.zip',
+    },
+    {
+        'source_urls': ['https://github.com/xflouris/libpll-2/archive/'],
+        'download_filename': local_libpll_tree + '.zip',
+        'filename': 'libpll-2.zip',
+    },
+]
+
+# copy dependencies into place
+preconfigopts = ("cp -a %(builddir)s/terraphast-one-" + local_terrap_tree +
+                 "/* %(builddir)s/raxml-ng-%(version)s/libs/terraphast/ && ")
```

Comment:
please move all `preconfigopts` to before `configopts`, for readability, and consider using formatted strings, e.g. `"cp -a %%(builddir)s/terraphast-one-%s/*" % local_terrap_commit`

Reply:
hm, the result is not quite what I meant, I meant move to `preconfigopts` down to before `configopts`, not `configopts` up to after `preconfigopts` (the reason is that we try to order the easyconfig in roughly the same order of the steps that EB takes)

so now I would move both `preconfigopts` and `configopts` to just after `dependencies`, before `files_to_copy` - but I understand you've already put a lot of work into this, and it looks good enough to me, so up to you

Reply:
No worries, I don't mind to update the pr, so we get it perfectly done.

But correct me if I'm wrong, the order would be starting from sources: checksums, dependencies, preconfigopts, configopts, build_type, separate_build, builddependencies, files_to_copy, sanity_check, and moduleclass.

So actually the clause out of order is dependencies, which should be just after checksums. 

Like this you have the phases: sources, preconfig, config, build, install, check.

Would it be like this?

Reply:
well, I see what you mean, but note for instance that builddependencies must be loaded before the config step, not only during the build step

in any case, the ordering is of course not strict, it's more about readability and where most people expect things to be